### ISSUE TITLE
MSTest 3.2.0 on Linux can't run the .NET Framework test .exe file with Permission Denied

### ISSUE THREAD

**Issue by AptiviCEO** [microsoft/testfx#2303]
## Describe the bug

In Linux and macOS systems, when trying to run `dotnet test` on a solution that contains a test project that targets both the modern .NET and the classic .NET Framework and has the following properties:

```xml
  <PropertyGroup>
    <TargetFrameworks>net8.0;net48</TargetFrameworks>
    <LangVersion>latest</LangVersion>
    <EnableMSTestRunner>true</EnableMSTestRunner>
    <OutputType>Exe</OutputType>
    <IsPackable>false</IsPackable>
    <IsTestProject>true</IsTestProject>
    <TestingPlatformDotnetTestSupport>true</TestingPlatformDotnetTestSupport>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="MSTest" Version="3.2.0" />
  </ItemGroup>
```

The .NET 8.0 version of the built tests runs successfully. As soon as it tries to run tests for the .NET Framework targeted version, it fails with:

```
  Run tests: '/home/runner/work/Terminaux/Terminaux/Terminaux.Tests/bin/Debug/net48/Terminaux.Tests.exe' [net48|x64]
/home/runner/.nuget/packages/microsoft.testing.platform.msbuild/1.0.0/buildMultiTargeting/Microsoft.Testing.Platform.MSBuild.targets(221,5): error MSB6003: The specified task executable "Terminaux.Tests.exe" could not be run. System.ComponentModel.Win32Exception (13): An error occurred trying to start process '/home/runner/work/Terminaux/Terminaux/Terminaux.Tests/bin/Debug/net48/Terminaux.Tests.exe' with working directory '/home/runner/work/Terminaux/Terminaux/Terminaux.Tests'. Permission denied [/home/runner/work/Terminaux/Terminaux/Terminaux.Tests/Terminaux.Tests.csproj::TargetFramework=net48]
/home/runner/.nuget/packages/microsoft.testing.platform.msbuild/1.0.0/buildMultiTargeting/Microsoft.Testing.Platform.MSBuild.targets(221,5): error MSB6003:    at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) [/home/runner/work/Terminaux/Terminaux/Terminaux.Tests/Terminaux.Tests.csproj::TargetFramework=net48]
/home/runner/.nuget/packages/microsoft.testing.platform.msbuild/1.0.0/buildMultiTargeting/Microsoft.Testing.Platform.MSBuild.targets(221,5): error MSB6003:    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) [/home/runner/work/Terminaux/Terminaux/Terminaux.Tests/Terminaux.Tests.csproj::TargetFramework=net48]
/home/runner/.nuget/packages/microsoft.testing.platform.msbuild/1.0.0/buildMultiTargeting/Microsoft.Testing.Platform.MSBuild.targets(221,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands) [/home/runner/work/Terminaux/Terminaux/Terminaux.Tests/Terminaux.Tests.csproj::TargetFramework=net48]
/home/runner/.nuget/packages/microsoft.testing.platform.msbuild/1.0.0/buildMultiTargeting/Microsoft.Testing.Platform.MSBuild.targets(221,5): error MSB6003:    at Microsoft.Build.Utilities.ToolTask.Execute() [/home/runner/work/Terminaux/Terminaux/Terminaux.Tests/Terminaux.Tests.csproj::TargetFramework=net48]
```

Full log here: https://github.com/Aptivi/Terminaux/actions/runs/7861057112/
macOS logs: https://github.com/Aptivi/Terminaux/actions/runs/7861057107/

## Steps To Reproduce

1. Create a solution that contains a unit test project
2. Open the unit test project file and copy the properties shown in the beginning of the report to the project file
3. Open the terminal to point to the directory that contains the solution file
4. Run `dotnet test` or `dotnet test --configuration Debug`

Repository: https://github.com/Aptivi/Terminaux

## Expected behavior

All tests, including the .NET Framework targeted build output, should run on Linux and macOS systems.

## Actual behavior

The test fails with this error:

```
/home/runner/.nuget/packages/microsoft.testing.platform.msbuild/1.0.0/buildMultiTargeting/Microsoft.Testing.Platform.MSBuild.targets(221,5): error MSB6003: The specified task executable "Terminaux.Tests.exe" could not be run. System.ComponentModel.Win32Exception (13): An error occurred trying to start process '/home/runner/work/Terminaux/Terminaux/Terminaux.Tests/bin/Debug/net48/Terminaux.Tests.exe' with working directory '/home/runner/work/Terminaux/Terminaux/Terminaux.Tests'. Permission denied [/home/runner/work/Terminaux/Terminaux/Terminaux.Tests/Terminaux.Tests.csproj::TargetFramework=net48]
```

[AB#1955197](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1955197)

**Comment by MarcoRossignoli** [comment]
Thanks for reporting this, we will exclude .NET Framework from non Windows system.

**Comment by Evangelink** [comment]
@AptiviCEO we just released 3.2.1 with a fix for the issue.

**Comment by AptiviCEO** [comment]
Thanks! The tests for .NET Framework 4.8 now run successfully!

Logs: https://github.com/Aptivi/Terminaux/actions/runs/7899485944/job/21559058820
